lib/streamlit/testing/element_tree.py

```diff
@@ -42,14 +46,15 @@ class Element:
     type: str
     proto: Any = field(repr=False)
     root: ElementTree = field(repr=False)
-    key: str | None = None
```

Comment:
@AnOctopus  keeping the default value None for the `key` field in `Element` class makes mypy complain about order of fields in dataclass.

Do we have any reason to keep the key field in `Element` in general, elements (the opposite of widgets) don't have a key. 
We could move it to mixin class, and use if we need it.

Reply:
I think it only has `key` to simplify the implementation of `Block.get_widget`. Which is easy to change to check if the element is a widget, if dropping `key` from `Element` would make other stuff easier.

Reply:
Good, I think we could merge this PR as is, and then maybe reorganize classes and fields there to have a more semantically correct inheritance hierarchy conforming to our models. I think we could discuss this later together!